run/pubsub/main.py

```diff
@@ -13,16 +13,24 @@
 # limitations under the License.
 
 # [START cloudrun_pubsub_server_setup]
-# [START run_pubsub_server_setup]
+# [START cloudrun_pubsub_server]
 import base64
 import os
 
 from flask import Flask, request
 
 
 app = Flask(__name__)
-# [END run_pubsub_server_setup]
+
+if __name__ == "__main__":
+    PORT = int(os.getenv("PORT")) if os.getenv("PORT") else 8080
+
+    # This is used when running locally. Gunicorn is used to run the
+    # application on Cloud Run. See entrypoint in Dockerfile.
+    app.run(host="127.0.0.1", port=PORT, debug=False)
+
```

Comment:
A python expert may weigh in but it may be more idiomatic to have this at the bottom of the file and duplicate the region tags around it

Reply:
I didn't encompass this in the description but that's what I was pairing with Nick on before. Proposed solution was the second best (first was creating a separate file).

Reply:
The if __name__ section is not being used in the sample. It's being launched by gunicorn in the Dockerfile and in Cloud Run. It would be better to remove it than move it.